wgengine/magicsock/endpoint.go

```diff
@@ -207,6 +207,26 @@ func (de *endpoint) deleteEndpointLocked(why string, ep netip.AddrPort) {
 	}
 }
 
+func (de *endpoint) clearEndpoints(why string) {
+	de.mu.Lock()
+	defer de.mu.Unlock()
+	de.debugUpdates.Add(EndpointChange{
+		When: time.Now(),
+		What: "clearEndpoints-" + why,
+	})
+	de.endpointState = map[netip.AddrPort]*endpointState{}
```

Comment:
I like this, but I think we should pair it with a lower-level implementation of `blockEndpoints`

Right now, we check this field on the magicsock `Conn` before calling into methods on `endpoint` that could add an endpoint.  There are 3 of these (`updateFromNode`, `handleCallMeMaybe` and `addCandidateEndpoint`), but you've only directly blocked the first two.

So, there is still a narrow race condition where you can set block endpoints, clear them all out, and a Disco Ping could come in and add an endpoint back in.

What do you think about extending `endpoint` to also have a `blockEndpoints` field that we sync with `Conn`, that way we can always directly check whether we are blocking endpoints at the moment we are about to add one?